CMakeLists.txt

```diff
@@ -148,6 +148,9 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin/")
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib/")
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib/")
 
+# Enable generating compile_commands.json
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
Prefer option instead set.
For building procedure, there is no need generate it

Reply:
There is no disadvantage of generating it, and it's useful for some IDEs and intellisense.

Reply:
This reasoning would more convince me if the recommended build platform would be the low-end device. However, most of the time the (cross)-build is performed on the host device where this extra json doesn't add to much for the "pollution" what cmake generates.

Reply:
@lygstate What exactly do you mean by pollution? The file is generated into the build tree which is already full of cmake-generated auxiliary files.

Reply:
> @lygstate What exactly do you mean by pollution? The file is generated into the build tree which is already full of cmake-generated auxiliary files.

The parent project also generating compile_commands.json

Reply:
@lygstate Is that a real issue anywhere?

@rerobika How about finding a middle ground and adding a cli option to build.py that controls this cmake variable?

Reply:
I can deal with that if it is enabled by default.

Reply:
option(CMAKE_EXPORT_COMPILE_COMMANDS "Enable export compile commands for jerryscript" ON)

Reply:
@lygstate Thanks, we _do_ know how to add an option to cmake. The suggestion was to add an option to `build.py` (we know how to add that, too). What we do not know whether the "pollution" is a real issue in any embedding project?